File: docs/snippets/python/error-content_main.py
# from microsoft.agents.xml.models import AgentMessage, ErrorContent
# from microsoft.agents.xml.serialization import MessageSerializer
#
# # Create message with error
# message = AgentMessage(
#     role="assistant",
#     agent_id="agent-456",
#     message_id="msg-error-1",
#     contents=[
#         ErrorContent(
#             code="rate_limit_exceeded",
#             message="Rate limit exceeded. Please try again in 60 seconds."
#         )
#     ]
# )
#
# serializer = MessageSerializer()
# xml_output = serializer.serialize(message)
# print(xml_output)

File: snippets/csharp/error-content_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;

// Create message with error
var message = new AgentMessage
{
    AgentId = "agent-456",
    MessageId = "msg-error-1",
    Contents = new List<AIContent>
    {
        new ErrorContent
        {
            Code = "rate_limit_exceeded",
            Message = "Rate limit exceeded. Please try again in 60 seconds."
        }
    }
};

var serializer = new MessageSerializer();
var xmlOutput = serializer.Serialize(message);
Console.WriteLine(xmlOutput);